scala/private/rule_impls.bzl

```diff
@@ -949,7 +832,7 @@ def scala_binary_common(
     )  # no need to build an ijar for an executable
     rjars = depset(outputs.full_jars, transitive = [rjars])
 
-    _build_deployable(ctx, rjars.to_list())
+    build_deployable(ctx, rjars.to_list())
```

Comment:
we should probably look at all the linearization of depsets we do in these rules and see if it is avoidable at all. It is a shame you can't lazily linearize a depset into args so it can happen at build time not analysis time... maybe you can.

Reply:
totally agree. I have a feeling our analysis perf can be greatly improved but I haven't gotten around to profiling it